vault/plugincatalog/plugin_artifact.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package plugincatalog
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/ProtonMail/gopenpgp/v3/crypto"
+)
+
+// hashiCorpPGPPubKey is HashiCorp's PGP public key at https://www.hashicorp.com/.well-known/pgp-key.txt.
+// This key is used to verify the authenticity of HashiCorp plugins.
+const hashiCorpPGPPubKey = `
```

Comment:
Should vault fetch this from the well-known endpoint and cache it? What happens if this ever needs to change? Ideally that would not happen, but it has happened in the past.